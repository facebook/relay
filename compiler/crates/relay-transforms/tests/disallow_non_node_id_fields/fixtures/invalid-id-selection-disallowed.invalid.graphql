# expected-to-throw
fragment UserWithDisallowedCustomIDType_fragment on UserWithDisallowedCustomIDType {
    id
    name
}

# %extensions%

scalar DisallowedCustomIDType

type UserWithDisallowedCustomIDType {
    id: DisallowedCustomIDType!
    name: String!
}