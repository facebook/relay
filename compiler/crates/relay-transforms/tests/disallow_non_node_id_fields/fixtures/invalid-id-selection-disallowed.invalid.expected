==================================== INPUT ====================================
# expected-to-throw
fragment UserWithDisallowedCustomIDType_fragment on UserWithDisallowedCustomIDType {
    id
    name
}

# %extensions%

scalar DisallowedCustomIDType

type UserWithDisallowedCustomIDType {
    id: DisallowedCustomIDType!
    name: String!
}
==================================== ERROR ====================================
✖︎ Disallowed type `DisallowedCustomIDType!` of field `id` on parent type `UserWithDisallowedCustomIDType` cannot be used by Relay to identify entities. For a detailed explanation, check out https://relay.dev/docs/debugging/disallowed-id-types-error

  <generated>:2:56
    1 │ # expected-to-throw
    2 │ fragment UserWithDisallowedCustomIDType_fragment on UserWithDisallowedCustomIDType {
      │                                                        ^^
    3 │     id
